include/net/cloud.h

```diff
@@ -281,6 +290,24 @@ static inline int cloud_input(const struct cloud_backend *const backend)
 	return backend->api->input(backend);
 }
 
+/**@brief Add list of cloud endpoint topics.
+ *
+ * @param backend Pointer to cloud backend structure.
+ * @param endpoint Pointer to list of endpoint topics.
+ * @param topics_count Number of endpoint topics in list
+ */
+static inline int cloud_ep_topics_set(const struct cloud_backend *const backend,
```

Comment:
Yes, it makes sense to keep away technology specific concepts in a generic API. In the spirit of generality I will change it to your proposal. I was under the assumption that the Cloud API would only be used with pub/sub cloud architectures, where "topics" are common.

Reply:
```cloud_ep_subscriptions_set``` is a little bit lengthy forcing all the input argument to have their own line. Even cloud_subscriptions_set is a little bit long. How about ```cloud_sub_list_set```?

Reply:
I'm fine with the shorter form, but let's keep this thread open for other proposals. Maybe someone will come up with some brilliant name for this API function.

Reply:
I think my lacking vocabulary when naming this `endpoints` in the first place is coming back to confuse us now. Endpoint in the cloud API context was originally intended to cover topics in MQTT, resources in CoaP, URI in REST based APIs and so on. So `_endpoints_topics_` would thus be sort of redundant naming, or at least specific for MQTT.

If I understand the intention of this function correctly, it's a way to configure endpoints for incoming traffic from a cloud. It would be possible to add endpoints on runtime, ie the list does not have to be exhaustive at compile time, and the function can be called multiple times without overwritin existing endpoints. If that understanding is correct, perhaps something like `cloud_ep_subscriptions_add` is a fit? We would need anouther `_remove` / `_delete` to go along with it, though.

Reply:
Is it really necessary with a corresponding delete functionality if all this can be done with the existing ```uninit``` API? I agree in the sense that it looks cleaner if ```cloud_ep_subscriptions_add``` had its own ```_remove```.

Reply:
What I had in mind for the `_delete` API was to, in MQTT context, unsubscribe from a topic to not waste data traffic on that topic if the device at some point doesn't need the data from the topic anymore

Reply:
So, an operation on a single endpoint, not wiping the entire list

Reply:
Perhaps not a super-important feature